packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/utils/common/bboxUtils.ts

```diff
@@ -56,6 +51,14 @@ export function isNextToEachOther(boxA: Bbox, boxB: Bbox): boolean {
   }
 
   // see if boxes can be neighborhoods
-  const verticalGap = Math.max(0, leftB - rightA, leftA - rightB);
-  return verticalGap < avgHeight;
+  const horizontalGap = Math.max(leftB - rightA, leftA - rightB);
+  if (horizontalGap < 0) {
+    // gap < 0 means that the boxes are overlapped
+    const overlap = -horizontalGap;
+    // consider two horizontally aligned overlapped boxes are next to each other when
+    // the overlap is smaller than half of the box height.
+    return overlap < avgHeight / 2;
+  }
```

Comment:
the previous code of disallowing any overlapping was too strict.
```
  if (bboxesIntersect(boxA, boxB)) {
    return false;
  }
```

Practically, PDF sometimes includes bboxes overlapped each other for a text. So, this code is to allow that kind of overlapping.

Reply:
I don't fully understand why you are using **height** to determine if boxes are **horizontally** aligned. That seems odd. What's the reasoning behind that?

Reply:
Not sure, but my guess is that this is for picking a small size that is relative to the size of the characters. (Width wouldn't work due to differences in word length.) My initial thought would be to pick a specific small size, but this solution better accounts for potential issues with zoom.

Reply:
This is to to adapt to various font sizes. For font height of 72px, 30px is small enough. But for height of 12px, 30px overlapping means that some characters are most likely overlapped. In the case, given boxes cannot be considered as 'next to each other'.